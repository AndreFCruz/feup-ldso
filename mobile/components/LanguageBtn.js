import React from 'react';
import { StyleSheet } from 'react-native';
import PropTypes from 'prop-types';
import { Button, Text } from 'native-base';
import { connect } from 'react-redux';
import { toggleLanguage } from '../reducers/modules/languageReducer';

class LanguageBtn extends React.Component {
    render() {
        const { language, invertedMode } = this.props;

        return (
            <Button transparent light={invertedMode} dark={!invertedMode} onPress={ () => this.props.toggleLanguage()}>
                <Text style={styles.button}>{language}</Text>
            </Button>
        );
    }
}

LanguageBtn.propTypes = {
    language: PropTypes.string,
    toggleLanguage: PropTypes.func,
    invertedMode: PropTypes.bool
};

const mapStateToProps = ({ language }) => ({
    language: language.selection
});

const mapDispatchToProps = {
    toggleLanguage
};

const styles = StyleSheet.create({
    button: {
        fontWeight: 'bold',
    },
});

export default connect(mapStateToProps, mapDispatchToProps)(LanguageBtn);
