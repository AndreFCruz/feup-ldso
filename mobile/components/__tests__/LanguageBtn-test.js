import 'react-native';
import React from 'react';
import LanguageBtn from '../LanguageBtn';
import Enzyme, { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

const middlewares = []; // you can mock any middlewares here if necessary
const mockStore = configureStore(middlewares);

function setup(invertedMode, language = 'en') {
    const initialState = { 
        language: {
            selection: language
        }
    };

    const wrapper = shallow(<LanguageBtn invertedMode={invertedMode} />,
        { context: { store: mockStore(initialState) } });

    return wrapper;
}

it('renders with black text', () => {
    const wrapper = setup(false);

    expect(wrapper.dive()).toMatchSnapshot();
});

it('renders with white text', () => {
    const wrapper = setup(true);

    expect(wrapper.dive()).toMatchSnapshot();
});

it('renders text correctly', () => {
    let wrapper = setup(true, 'en');

    let render = wrapper.dive();
    expect(render.find('Styled(Text)').contains('EN')).toEqual(true);

    wrapper = setup(true, 'pt');

    render = wrapper.dive();
    expect(render.find('Styled(Text)').contains('PT')).toEqual(true);
});
