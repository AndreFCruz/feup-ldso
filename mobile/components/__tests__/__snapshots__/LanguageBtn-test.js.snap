// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with black text 1`] = `
<Styled(Button)
  dark={true}
  light={false}
  onPress={[Function]}
  transparent={true}
>
  <Styled(Text)
    style={
      Object {
        "fontWeight": "bold",
      }
    }
  >
    en
  </Styled(Text)>
</Styled(Button)>
`;

exports[`renders with white text 1`] = `
<Styled(Button)
  dark={false}
  light={true}
  onPress={[Function]}
  transparent={true}
>
  <Styled(Text)
    style={
      Object {
        "fontWeight": "bold",
      }
    }
  >
    en
  </Styled(Text)>
</Styled(Button)>
`;
